upgrade/boot_loader/VerifierEcDsa.hpp

```diff
@@ -13,9 +15,43 @@ namespace application
 
         bool IsValid(hal::SynchronousFlash& flash, const hal::SynchronousFlash::Range& signature, const hal::SynchronousFlash::Range& data) const override;
 
+    protected:
+        virtual infra::ConstByteRange GetSignatureStorage(const hal::SynchronousFlash::Range& signature) const = 0;
+        virtual uECC_Curve GetCurve() const = 0;
+
     private:
         infra::ConstByteRange key;
     };
+
+    class VerifierEcDsa224
+        : public VerifierEcDsa
+    {
+    public:
+        explicit VerifierEcDsa224(infra::ConstByteRange key);
+
+    protected:
+        infra::ConstByteRange GetSignatureStorage(const hal::SynchronousFlash::Range& signature) const override;
+        uECC_Curve GetCurve() const override;
+
+    private:
+        static constexpr size_t signSize = 224 / 8 * 2;
+        std::array<uint8_t, signSize> signatureStorage;
```

Comment:
signatureStorage is not part of the externally observable state of VerifierEcDsa; it is merely used as scratchpad in an operation that does not modify observable state. This value should therefore be mutable.

```suggestion
        mutable std::array<uint8_t, signSize> signatureStorage;
```